apps/opencs/view/render/terrainselection.cpp

```diff
@@ -0,0 +1,251 @@
+    const int landSize {ESM::Land::LAND_SIZE};
+    const int landTextureSize {ESM::Land::LAND_TEXTURE_SIZE};
+}
+
+CSVRender::TerrainSelection::TerrainSelection(osg::Group* parentNode, WorldspaceWidget *worldspaceWidget):
+mParentNode(parentNode), mWorldspaceWidget (worldspaceWidget)
+{
+    osg::ref_ptr<osg::Geometry> geometry = new osg::Geometry();
+
+    // Color
+    osg::ref_ptr<osg::Vec4Array> colors = new osg::Vec4Array{};
+    colors->push_back(osg::Vec4f(0.f, 0.5f, 0.f, 1.f));
+
+    geometry->setColorArray(colors, osg::Array::Binding::BIND_OVERALL);
+
+    // Primitive
+    osg::ref_ptr<osg::DrawArrays> drawArrays = new osg::DrawArrays{osg::PrimitiveSet::Mode::LINES};
+
+    geometry->addPrimitiveSet(drawArrays);
+    geometry->getOrCreateStateSet()->setMode(GL_LIGHTING, osg::StateAttribute::Values::OFF);
+
+    mGeode = new osg::Geode();
+    mGeode->addDrawable(geometry);
+
+    activate();
+}
+
+CSVRender::TerrainSelection::~TerrainSelection()
+{
+    deactivate();
+}
+
+std::vector<std::pair<int, int>> CSVRender::TerrainSelection::getTerrainSelection() const
+{
+    return mSelection;
+}
+
+void CSVRender::TerrainSelection::selectTerrainTexture(const WorldspaceHitResult& hit)
+{
+    mSelection.clear();
+
+    const std::pair<int, int> localPos {toTextureCoords(hit.worldPos)};
+
+    if (std::find(mSelection.begin(), mSelection.end(), localPos) == mSelection.end())
+        mSelection.push_back(localPos);
+
+    update();
+}
+
+void CSVRender::TerrainSelection::onlyAddSelect(const WorldspaceHitResult& hit)
+{
+    const std::pair<int, int> localPos {toTextureCoords(hit.worldPos)};
+
+    if (std::find(mSelection.begin(), mSelection.end(), localPos) == mSelection.end())
+            mSelection.push_back(localPos);
+
+    update();
+}
+
+void CSVRender::TerrainSelection::toggleSelect(const WorldspaceHitResult& hit)
+{
+    const std::pair<int, int> localPos {toTextureCoords(hit.worldPos)};
+    const auto iter = std::find(mSelection.begin(), mSelection.end(), localPos);
+
+    if (iter != mSelection.end()) {
+        mSelection.erase(iter);
+    }
+    else {
+        mSelection.push_back(localPos);
+    }
+    update();
+}
+
+void CSVRender::TerrainSelection::activate()
+{
+    mParentNode->addChild(mGeode);
+}
+
+void CSVRender::TerrainSelection::deactivate()
+{
+    mParentNode->removeChild(mGeode);
+}
+
+std::pair<int, int> CSVRender::TerrainSelection::toTextureCoords(osg::Vec3d worldPos) const
+{
+    const double xd {worldPos.x() * landTextureSize / cellSize+(landTextureSize / cellSize)/4};
+    const double yd {worldPos.y() * landTextureSize / cellSize+(landTextureSize / cellSize)/4};
+
+    const auto x = static_cast<int>(std::floor(xd));
+    const auto y = static_cast<int>(std::floor(yd));
+
+    return std::make_pair(x, y);
+}
+
+std::pair<int, int> CSVRender::TerrainSelection::toVertexCoords(osg::Vec3d worldPos) const
+{
+    const double xd {worldPos.x() * landTextureSize / cellSize+(landTextureSize / cellSize)/4};
```

Comment:
I suggest we avoid this method of initialising variables and stay with the = again. There aren't any real advantages to {} and we are just introducing new inconsistencies into the code base.

Reply:
Btw, is initialising with () okay? In commit 80ee08c I had an oversight and changed {} to (), although you specifically wrote =.